src/metadata/mod.rs

```diff
@@ -0,0 +1,58 @@
+use elasticsearch::cluster::ClusterHealthParts;
+use elasticsearch::nodes::NodesInfoParts;
+use elasticsearch::{Elasticsearch, Error};
+use serde_json::Value;
+use std::collections::HashMap;
+
+type NodeIdToName = HashMap<String, String>;
+
+pub(crate) async fn nodes_id_map(client: &Elasticsearch) -> Result<NodeIdToName, Error> {
+    let nodes_os = client
+        .nodes()
+        .info(NodesInfoParts::None)
+        // TODO: fix/report bug for filter_path
+        // DEBUG reqwest::async_impl::client > response '200 OK' for http://supernode:9200/_nodes?filter_path=os
+        // ERROR elasticsearch_exporter      > error decoding response body: missing field `nodes` at line 1 column 2
+        // .filter_path(&["os"])
+        .send()
+        .await?
+        .json::<NodesOs>()
+        .await?;
+
+    Ok(nodes_os.into())
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NodesOs {
+    nodes: HashMap<String, NodeData>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NodeData {
+    name: String,
+}
+
+impl From<NodesOs> for HashMap<String, String> {
+    fn from(node_os: NodesOs) -> Self {
+        let mut map = NodeIdToName::new();
```

Comment:
Would it be possible to rewrite this avoiding mutable variables, e.g. using `.map`?

Reply:
Maybe, this could also use `serde_json::Map` but we would lose typing